yt_dlp/extractor/ted.py

```diff
@@ -149,84 +90,76 @@ def _real_extract(self, url):
+        series = try_get(json, lambda x: x['props']['pageProps']['series'])
+        series_id = series.get('id')
 
-        info = self._extract_info(webpage)
+        seasonlist = try_get(json, lambda x: x['props']['pageProps']['seasons'], list)
 
-        data = try_get(info, lambda x: x['__INITIAL_DATA__'], dict) or info
-        talk_info = data['talks'][0]
-
-        title = talk_info['title'].strip()
-
-        downloads = talk_info.get('downloads') or {}
-        native_downloads = downloads.get('nativeDownloads') or talk_info.get('nativeDownloads') or {}
-
-        formats = [{
-            'url': format_url,
-            'format_id': format_id,
-        } for (format_id, format_url) in native_downloads.items() if format_url is not None]
-
-        subtitled_downloads = downloads.get('subtitledDownloads') or {}
-        for lang, subtitled_download in subtitled_downloads.items():
-            for q in self._NATIVE_FORMATS:
-                q_url = subtitled_download.get(q)
-                if not q_url:
-                    continue
-                formats.append({
-                    'url': q_url,
-                    'format_id': '%s-%s' % (q, lang),
-                    'language': lang,
-                })
-
-        if formats:
-            for f in formats:
-                finfo = self._NATIVE_FORMATS.get(f['format_id'].split('-')[0])
-                if finfo:
-                    f.update(finfo)
-
-        player_talk = talk_info['player_talks'][0]
+        playlist_entries = []
+        if season:
+            for s in [playlist for playlist in seasonlist if playlist['seasonNumber'] == season]:
+                playlist_entries.extend(self._parse_playlist(s))
+        else:
+            playlist_entries = self._parse_playlist(seasonlist[0])
 
-        resources_ = player_talk.get('resources') or talk_info.get('resources')
+        return self.playlist_result(
+            playlist_entries, playlist_id=series_id,
+            playlist_title=self._og_search_title(webpage, fatal=False),
+            playlist_description=self._og_search_description(webpage))
 
+    def _talk_info(self, url, video_name):
+        webpage = self._download_webpage(url, video_name)
+        json = self._extract_info(webpage, video_name)
+        talk_info = try_get(json, lambda x: x['props']['pageProps']['videoData'], dict)
+
+        video_id = talk_info.get('id')
+        title = talk_info.get('title') or self._og_search_title(webpage)
+        uploader = talk_info.get('presenterDisplayName')
+        release_date = unified_strdate(talk_info.get('recordedOn'))
+        upload_date = unified_strdate(talk_info.get('publishedAt'))
+        view_count = str_to_int(talk_info.get('viewedCount'))
+        duration = parse_duration(talk_info.get('duration')) or parse_duration(self._og_search_property('video:duration', webpage))
+        description = parse_duration(talk_info.get('description')) or self._og_search_description(webpage)
```

Comment:
do u really need all these vars, or can u inline a lot of then in the return (or atleast move them close to return)